juju/juju
# raft: Don't remove all servers in the clusterer

## Description of change

Sometimes when the leader is removed, the peer grouper reports no API servers with addresses. This prompted the raft clusterer to try to remove all servers and the raft cluster would get into a bad state. Avoid this with a sanity check - discard updates where all machines have been removed at once.

## QA steps

Bootstrap a 3-node controller.
Remove the leader (probably machine-0).
Ensure that the raft cluster is as expected by running juju-engine-report on both of the remaining machines.
